### PR TITLE
Styling and layout changes to DeviceDetailsPage and minor changes to DeviceList

### PR DESCRIPTION
**Pull request by sharon-ho**
# General info

**Issue number**: N/A

**Task description**: 
- Removing redundant Overview table/tab from the rightmost table in the page
- Add an overview tab to the leftmost table and remove the click functionality to the tab
- Change edit button into an Edit icon and place it next to the Device Name, since it will be the only editable property
- Move Delete button to be on the same line as Device Name
- Extremely minor changes: changed indicator color in DeviceList vertical tabs and added padding to the DetailsPanel in DeviceList
 
 
| Original   | Updated|
|:---------------:|:---------------:|
| image | image |

# Testing

**Test coverage**: 100%

# Additional notes

**Note to reviewers**:
N/A

**To do before merging**:
- [x] Raise code coverage
- [x] Wait for #248 to be merged first

### CODE REVIEW

File: frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx
Comment (jun-loke) [inline frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx]: I don't think you need to have the tab here. since this doesn't use the tab functionality. I reckon you can just leave it as a `DeviceDetailsConciseTable` component only.

File: frontend/src/DeviceDetailsPage/DeviceDetailsTabTable.jsx
Comment (jun-loke) [inline frontend/src/DeviceDetailsPage/DeviceDetailsTabTable.jsx]: Why did you remove Paper?

File: frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx
Comment (jun-loke) [inline frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx]: Any particular reasonw e're using thruthy & falsy instead of true and false?

File: frontend/src/api/DeviceApi.js
Comment (jun-loke) [inline frontend/src/api/DeviceApi.js]: If you aren't doing anything with the error, I don't think you need to put in that argument in the lambda function

File: frontend/src/devicelist/ActionMenu.jsx
Comment (jun-loke) [inline frontend/src/devicelist/ActionMenu.jsx]: Not used. Remove

**Review by jun-loke — CHANGES_REQUESTED** [review]
Just some questions and clarifications

File: frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx
Comment (sharon-ho) [reply frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx]: Not related to my PR, so it would be better to address this issue in @gdzooks's #248 PR, since these changes were added because I branched off their feature to work on this PR.

File: frontend/src/api/DeviceApi.js
Comment (sharon-ho) [reply frontend/src/api/DeviceApi.js]: Not related to my PR, so it would be better to address this issue in @gdzooks's #248 PR, since these changes were added because I branched off their feature to work on this PR.

File: frontend/src/devicelist/ActionMenu.jsx
Comment (sharon-ho) [reply frontend/src/devicelist/ActionMenu.jsx]: Not related to my PR, so it would be better to address this issue in @gdzooks's #248 PR, since these changes were added because I branched off their feature to work on this PR.

File: frontend/src/DeviceDetailsPage/DeviceDetailsTabTable.jsx
Comment (sharon-ho) [reply frontend/src/DeviceDetailsPage/DeviceDetailsTabTable.jsx]: I removed it during manual testing to see if it was aesthetically nicer without, but then I re-added them to the individual TabPanel components instead by accident.

File: frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx
Comment (sharon-ho) [reply frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx]: Indeed there is nothing to "tab through", however I'm using the Tab component for aesthetic/consistency reasons to go alongside the TabPanels on the right hand side.

**Comment by sharon-ho** [comment]
@arejayelle Just letting you know this PR is currently a draft and I am waiting for @gdzooks's #248 PR to merge to master first so that GitHub removes their changes related to deleting a device from this PR when comparing files changed.

File: frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx
Comment (sharon-ho) [reply frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx]: image

File: frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx
Comment (gdzooks) [reply frontend/src/DeviceDetailsPage/__tests__/DeleteDeviceButton.test.jsx]: yeah, setOpen is a function, so it can't be a boolean value. truthy/false checks that a function is returning values that qualify as true/false respectively.

File: frontend/src/api/DeviceApi.js
Comment (gdzooks) [reply frontend/src/api/DeviceApi.js]: ideally that's where we'd put error messages n stuff but you're right that it's doing nothing rn. i'll remove it from StreamApi as well since i did the same thing there

File: frontend/src/DeviceDetailsPage/DeviceDetailsTabTable.jsx
Comment (sharon-ho) [reply frontend/src/DeviceDetailsPage/DeviceDetailsTabTable.jsx]: Actually I tested it out and adding the Paper component to each TabPanel is the right approach.

**If unchanged from before the PR:**
image

**If removed from DeviceDetailsTabTable but added individually to each TabPanel:**
image

**Review by gdzooks — APPROVED** [review]
i'm not a huge fan of the overview tab still looking like a button--even if it isn't clickable--but that's just a stylistic preference, and i think you've done other much better and more significant changes here! so you get my approval regardless

File: frontend/src/general/VerticalTabs.jsx
Comment (FGRCL) [inline frontend/src/general/VerticalTabs.jsx:36]: I think that color shows up as purple on my browser. Is that our primary color?
image

**Review by FGRCL — APPROVED** [review]
The listed changes look O.K. on my end

File: frontend/src/general/VerticalTabs.jsx
Comment (sharon-ho) [reply frontend/src/general/VerticalTabs.jsx:36]: Yeah that color is set to our primary, I think it's the default one by Material UI.

**Review by jun-loke — APPROVED** [review]
Most of the changes were addressed in the different PR

**Pull request merged**
